x-pack/plugins/security_solution/public/assistant/provider.tsx

```diff
@@ -7,51 +7,114 @@
 import React, { useCallback } from 'react';
 import { i18n } from '@kbn/i18n';
 import type { IToasts } from '@kbn/core-notifications-browser';
-import { AssistantProvider as ElasticAssistantProvider } from '@kbn/elastic-assistant';
+import type { Conversation } from '@kbn/elastic-assistant';
+import {
+  AssistantProvider as ElasticAssistantProvider,
+  bulkChangeConversations,
+  mergeBaseWithPersistedConversations,
+  useFetchCurrentUserConversations,
+} from '@kbn/elastic-assistant';
 
+import type { FetchConversationsResponse } from '@kbn/elastic-assistant/impl/assistant/api';
+import { once } from 'lodash/fp';
 import { useBasePath, useKibana } from '../common/lib/kibana';
 import { useAssistantTelemetry } from './use_assistant_telemetry';
 import { getComments } from './get_comments';
-import { augmentMessageCodeBlocks, LOCAL_STORAGE_KEY } from './helpers';
-import { useConversationStore } from './use_conversation_store';
+import { LOCAL_STORAGE_KEY, augmentMessageCodeBlocks } from './helpers';
+import { useBaseConversations } from './use_conversation_store';
 import { DEFAULT_ALLOW, DEFAULT_ALLOW_REPLACEMENT } from './content/anonymization';
 import { PROMPT_CONTEXTS } from './content/prompt_contexts';
 import { BASE_SECURITY_QUICK_PROMPTS } from './content/quick_prompts';
 import { BASE_SECURITY_SYSTEM_PROMPTS } from './content/prompts/system';
 import { useAnonymizationStore } from './use_anonymization_store';
 import { useAssistantAvailability } from './use_assistant_availability';
-import { APP_ID } from '../../common/constants';
 import { useAppToasts } from '../common/hooks/use_app_toasts';
 import { useSignalIndex } from '../detections/containers/detection_engine/alerts/use_signal_index';
 
 const ASSISTANT_TITLE = i18n.translate('xpack.securitySolution.assistant.title', {
   defaultMessage: 'Elastic AI Assistant',
 });
 
+const LOCAL_CONVERSATIONS_MIGRATION_STATUS_TOAST_TITLE = i18n.translate(
+  'xpack.securitySolution.assistant.conversationMigrationStatus.title',
+  {
+    defaultMessage: 'Local storage conversations persisted successfuly.',
+  }
+);
+
 /**
  * This component configures the Elastic AI Assistant context provider for the Security Solution app.
  */
 export const AssistantProvider: React.FC = ({ children }) => {
   const {
     http,
+    notifications,
+    storage,
     triggersActionsUi: { actionTypeRegistry },
     docLinks: { ELASTIC_WEBSITE_URL, DOC_LINK_VERSION },
   } = useKibana().services;
   const basePath = useBasePath();
 
-  const { conversations, setConversations } = useConversationStore();
-  const getInitialConversation = useCallback(() => {
-    return conversations;
-  }, [conversations]);
-
+  const baseConversations = useBaseConversations();
   const assistantAvailability = useAssistantAvailability();
   const assistantTelemetry = useAssistantTelemetry();
 
+  const migrateConversationsFromLocalStorage = once(
```

Comment:
Do we want corresponding functions for migrating existing prompts & anonymization settings as well? I just performed an upgrade test and these are the results I'm seeing:

* Any existing custom System/Quick prompts are not migrated over and are not displayed in the UI. They still exist however in local storage under `securitySolution.securityAssistant.quickPrompts` and `securitySolution.securityAssistant.systemPrompts`, but don't show up in UI or in new `.kibana-elastic-ai-assistant-prompts-default` datastream
* Anonymization settings appear to migrate without issue, however they seems to still be using the settings stored in `securitySolution.securityAssistant.defaultAllowReplacement` and `securitySolution.securityAssistant.defaultAllow`. Just as with the prompts, the `.kibana-elastic-ai-assistant-anonymization-fields-default` datastream is empty, even after updating anonymization settings and saving (only local storage appears to be updated). That said, looks like replacements are being created and attached as expected in the conversations datastream. 


Here are the local storage dumps from before/after the upgrade test. Gist is `securitySolution.securityAssistant` key is removed post-upgrade (as convos stored in datastream now), and all other keys remain with three new keys being added: `elasticAssistantDefault.knowledgeBase`, `elasticAssistantDefault.quickPrompts`, and `elasticAssistantDefault.systemPrompts`.


<details><summary>Local Storage Dump Before Upgrade</summary>
<p>

```json
{
  "securitySolution.securityAssistant.systemPrompts": [
    {
      "id": "default-system-prompt",
      "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:",
      "name": "Default system prompt",
      "promptType": "system",
      "isDefault": true,
      "isNewConversationDefault": false
    },
    {
      "id": "CB9FA555-B59F-4F71-AFF9-8A891AC5BC28",
      "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nProvide the most detailed and relevant answer possible, as if you were relaying this information back to a cyber security expert.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:",
      "name": "Enhanced system prompt",
      "promptType": "system",
      "isDefault": true,
      "isNewConversationDefault": false
    },
    {
      "id": "New Pre-Persistence System Prompt",
      "content": "Body!",
      "name": "New Pre-Persistence System Prompt",
      "promptType": "system",
      "isNewConversationDefault": true
    }
  ],
  "securitySolution.securityAssistant": {
    "Alert summary": {
      "id": "Alert summary",
      "isDefault": true,
      "messages": [
        {
          "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:\\n\\nCONTEXT:\\n'''\\n@timestamp,2024-03-06T21:46:00.104Z\\n_id,3576558d-0937-4407-baca-ea53177b8218\\nevent.action,start\\nevent.category,f8b82f66-17e5-4914-83cd-c3f8add7913e\\nevent.dataset,69083092-afd6-4129-8188-5c51f341eff3\\nevent.module,endpoint\\nevent.type,info\\nhost.name,830262c9-adc6-4633-a90f-91c4f38f337e\\nkibana.alert.last_detected,2024-03-06T21:46:00.115Z\\nkibana.alert.risk_score,47\\nkibana.alert.rule.description,Generates a detection alert each time an Elastic Endpoint Security alert is received. Enabling this rule allows you to immediately begin investigating your Endpoint alerts.\\nkibana.alert.rule.name,Endpoint Security [Duplicate]\\nkibana.alert.severity,medium\\nkibana.alert.workflow_status,open\\nprocess.Ext.token.integrity_level_name,high\\nprocess.entity_id,vcnbcdc8go\\nprocess.executable,C:/fake/powershell.exe\\nprocess.hash.md5,fake md5\\nprocess.hash.sha1,fake sha1\\nprocess.hash.sha256,fake sha256\\nprocess.name,powershell.exe\\nprocess.parent.entity_id,6kqyqxsatx\\nprocess.pid,2\\nuser.domain,cd659a7c-101e-447e-89ad-ac01a86be78e\\nuser.name,bde5eed3-ee6f-424a-94ce-c10017d27f9f\\n'''\\n\\nAs an expert in security operations and incident response, provide a breakdown of the attached alert and summarize what it might mean for my organization.",
          "role": "user",
          "timestamp": "3/6/2024, 2:46:44 PM"
        },
        {
          "role": "assistant",
          "content": "The alert data provided contains several key elements that are crucial for understanding its implications for your organization. Here is a breakdown of the important fields and what they might mean:\\n\\n1. **Timestamp (`@timestamp`):** The alert was generated on March 6, 2024, at 21:46:00.104Z. This gives you the exact time when the event that triggered this alert occurred.\\n\\n2. **Event Information:**\\n   - **Action (`event.action`):** The event that caused the alert was a 'start' action. This could indicate the initiation of a process or a service.\\n   - **Category (`event.category`):** The identifier `f8b82f66-17e5-4914-83cd-c3f8add7913e` is unique and would typically relate to a specific category in your event categorization system, which could provide insight into the type of event (e.g., authentication, process start).\\n   - **Dataset (`event.dataset`):** The dataset `69083092-afd6-4129-8188-5c51f341eff3` could represent a specific source of events within the Elastic Security ecosystem, possibly relating to endpoint security.\\n   - **Module (`event.module`):** Indicates the alert comes from the 'endpoint' module, suggesting it is related to an endpoint security event.\\n\\n3. **Host Information (`host.name`):** The unique identifier `830262c9-adc6-4633-a90f-91c4f38f337e` represents the host from which the alert was generated. This helps in identifying the specific machine in your network that is affected or involved in the event.\\n\\n4. **Alert Details:**\\n   - **Risk Score (`kibana.alert.risk_score`):** The alert has been assigned a risk score of 47, suggesting a medium level of risk to your organization.\\n   - **Severity (`kibana.alert.severity`):** The severity level is marked as 'medium,' which indicates that the alert should be taken seriously but does not represent an immediate critical threat.\\n   - **Description (`kibana.alert.rule.description`):** The alert is generated as a result of receiving an Elastic Endpoint Security alert, aimed at facilitating immediate investigation.\\n   - **Rule Name (`kibana.alert.rule.name`):** Labeled as 'Endpoint Security [Duplicate]', suggesting it may be part of a standard set of rules for detecting duplicate endpoint security alerts.\\n\\n5. **Process Information:**\\n   - **Executable (`process.executable`):** The executable path is `C:/fake/powershell.exe`, suggesting that PowerShell was launched, which is common in both legitimate administrative tasks and malicious activities for script execution.\\n   - **Process Name (`process.name`):** The process `powershell.exe` confirms that the alert relates to the start of a PowerShell instance.\\n   - **Entity ID (`process.entity_id`):** Unique ID `vcnbcdc8go` for the process allows for tracking and correlation with other events or alerts.\\n\\n6. **User Information:**\\n   - **Domain (`user.domain`):** The domain `cd659a7c-101e-447e-89ad-ac01a86be78e` indicates which domain the user associated with the process belongs to, providing context for further investigation.\\n   - **Name (`user.name`):** The unique identifier `bde5eed3-ee6f-424a-94ce-c10017d27f9f` represents the user name, which may be used to trace back to an individual in your organization or an automated process/account.\\n\\n**Implications for Your Organization:**\\n\\nThe alert indicates that a PowerShell executable was started on a host within your organization with a medium risk score and severity. PowerShell usage is commonplace but can also be a vector for malicious activities, including executing scripts that perform unauthorized actions, such as data exfiltration or downloading malware.\\n\\nGiven the medium risk and severity, and the general use of PowerShell for both legitimate and illegitimate purposes, it is essential to investigate:\\n- The context in which PowerShell was launched.\\n- The scripts or commands that were executed.\\n- The user account that initiated the process and whether it was authorized to perform such actions.\\n- Network activity and file actions performed by the PowerShell process to identify any potential malicious behavior.\\n\\nThis alert should prompt a review of your organization's security posture, particularly regarding the management and monitoring of powerful tools like PowerShell, and could lead to tightening of controls and policies to prevent misuse.",
          "timestamp": "3/6/2024, 2:47:37 PM",
          "isError": false
        }
      ],
      "apiConfig": {
        "defaultSystemPromptId": "New Pre-Persistence System Prompt",
        "connectorId": "azure-open-ai",
        "connectorTypeTitle": "OpenAI"
      },
      "replacements": {
        "f8b82f66-17e5-4914-83cd-c3f8add7913e": "malware",
        "bde5eed3-ee6f-424a-94ce-c10017d27f9f": "su9b1wt9y2",
        "cd659a7c-101e-447e-89ad-ac01a86be78e": "61m9ydseun",
        "830262c9-adc6-4633-a90f-91c4f38f337e": "Host-o3wkipe7zq",
        "69083092-afd6-4129-8188-5c51f341eff3": "endpoint",
        "3576558d-0937-4407-baca-ea53177b8218": "980bfbe9ee905226ea493b6507ca719b1726aaeffe6ad12120ddcf76260bb098"
      }
    },
    "Data Quality dashboard": {
      "id": "Data Quality dashboard",
      "isDefault": true,
      "messages": [],
      "apiConfig": {}
    },
    "Detection Rules": {
      "id": "Detection Rules",
      "isDefault": true,
      "messages": [],
      "apiConfig": {}
    },
    "Event summary": {
      "id": "Event summary",
      "isDefault": true,
      "messages": [],
      "apiConfig": {}
    },
    "Timeline": {
      "excludeFromLastConversationStorage": true,
      "id": "Timeline",
      "isDefault": true,
      "messages": [],
      "apiConfig": {}
    },
    "Welcome": {
      "id": "Welcome",
      "isDefault": true,
      "theme": {
        "title": "Elastic AI Assistant",
        "titleIcon": "logoSecurity",
        "assistant": {
          "name": "Elastic AI Assistant",
          "icon": "logoSecurity"
        },
        "system": {
          "icon": "logoElastic"
        },
        "user": {}
      },
      "messages": [],
      "apiConfig": {
        "connectorId": "azure-open-ai",
        "connectorTypeTitle": "OpenAI"
      }
    }
  },
  "securitySolution.securityAssistant.lastConversationId": "Alert summary",
  "securitySolution.securityAssistant.defaultAllow": [
    "_id",
    "@timestamp",
    "cloud.availability_zone",
    "cloud.provider",
    "cloud.region",
    "destination.ip",
    "dns.question.name",
    "dns.question.type",
    "event.action",
    "event.category",
    "event.dataset",
    "event.module",
    "event.outcome",
    "event.type",
    "file.Ext.original.path",
    "file.hash.sha256",
    "file.name",
    "file.path",
    "host.name",
    "host.risk.calculated_level",
    "host.risk.calculated_score_norm",
    "kibana.alert.last_detected",
    "kibana.alert.risk_score",
    "kibana.alert.rule.description",
    "kibana.alert.rule.name",
    "kibana.alert.rule.references",
    "kibana.alert.rule.threat.framework",
    "kibana.alert.rule.threat.tactic.id",
    "kibana.alert.rule.threat.tactic.name",
    "kibana.alert.rule.threat.tactic.reference",
    "kibana.alert.rule.threat.technique.id",
    "kibana.alert.rule.threat.technique.name",
    "kibana.alert.rule.threat.technique.reference",
    "kibana.alert.rule.threat.technique.subtechnique.id",
    "kibana.alert.rule.threat.technique.subtechnique.name",
    "kibana.alert.rule.threat.technique.subtechnique.reference",
    "kibana.alert.severity",
    "kibana.alert.workflow_status",
    "process.args",
    "process.command_line",
    "process.executable",
    "process.Ext.token.integrity_level_name",
    "process.entity_id",
    "process.exit_code",
    "process.hash.md5",
    "process.hash.sha1",
    "process.name",
    "process.hash.sha256",
    "process.parent.args",
    "process.parent.args_count",
    "process.parent.code_signature.exists",
    "process.parent.code_signature.status",
    "process.parent.code_signature.subject_name",
    "process.parent.code_signature.trusted",
    "process.parent.command_line",
    "process.parent.entity_id",
    "process.parent.executable",
    "process.pid",
    "process.working_directory",
    "network.protocol",
    "source.ip",
    "user.domain",
    "user.name",
    "user.risk.calculated_level",
    "user.risk.calculated_score_norm"
  ],
  "securitySolution.securityAssistant.defaultAllowReplacement": [
    "_id",
    "cloud.availability_zone",
    "cloud.provider",
    "cloud.region",
    "destination.ip",
    "file.Ext.original.path",
    "file.name",
    "file.path",
    "host.ip",
    "host.name",
    "source.ip",
    "user.domain",
    "user.name"
  ],
  "securitySolution.securityAssistant.knowledgeBase": {
    "isEnabledRAGAlerts": false,
    "isEnabledKnowledgeBase": false,
    "latestAlerts": 20
  },
  "securitySolution.securityAssistant.quickPrompts": [
    {
      "title": "Alert summarization",
      "prompt": "As an expert in security operations and incident response, provide a breakdown of the attached alert and summarize what it might mean for my organization.",
      "color": "#F68FBE",
      "categories": [
        "alert"
      ],
      "isDefault": true
    },
    {
      "title": "ES|QL Query Generation",
      "prompt": "As an expert user of Elastic Security, please generate an accurate and valid ESQL query to detect the use case below. Your response should be formatted to be able to use immediately in an Elastic Security timeline or detection rule. Take your time with the answer, check your knowledge really well on all the functions I am asking for. For ES|QL answers specifically, you should only ever answer with what's available in your private knowledge. I cannot afford for queries to be inaccurate. Assume I am using the Elastic Common Schema and Elastic Agent.\\n\\nEnsure the answers are formatted in a way which is easily copyable as a separate code block in markdown.",
      "color": "#9170B8",
      "categories": [
        "knowledge-base"
      ],
      "isDefault": true
    },
    {
      "title": "Query generation",
      "prompt": "As an expert user of Elastic Security, please generate an accurate and valid EQL query to detect the use case below. Your response should be formatted to be able to use immediately in an Elastic Security timeline or detection rule. If Elastic Security already has a prebuilt rule for the use case, or a similar one, please provide a link to it and describe it.",
      "categories": [
        "detection-rules"
      ],
      "color": "#7DDED8",
      "isDefault": true
    },
    {
      "title": "Workflow suggestions",
      "prompt": "As an expert user of Elastic Security, please suggest a workflow, with step by step instructions on how to:",
      "color": "#36A2EF",
      "isDefault": true
    },
    {
      "title": "Custom data ingestion helper",
      "prompt": "As an expert user of Elastic Security, Elastic Agent, and Ingest pipelines, please list accurate and formatted, step by step instructions on how to ingest the following data using Elastic Agent and Fleet in Kibana and convert it to the Elastic Common Schema:",
      "categories": [
        "event"
      ],
      "color": "#F3D371",
      "isDefault": true
    },
    {
      "title": "Query conversion",
      "prompt": "I have the following query from a previous SIEM platform. As an expert user of Elastic Security, please suggest an Elastic EQL equivalent. I should be able to copy it immediately into an Elastic security timeline.",
      "color": "#BADA55",
      "isDefault": true
    },
    {
      "title": "Agent integration advice",
      "prompt": "Which Fleet enabled Elastic Agent integration should I use to collect logs and events from:",
      "color": "#FFA500",
      "isDefault": true
    },
    {
      "title": "Custom Pre-Persistence QP",
      "prompt": "No wonder there's so much body, you've got an organ going!",
      "color": "#D36086",
      "categories": [
        "detection-rules"
      ]
    },
    {
      "title": "Custom Pre-Persistence ",
      "prompt": "More body, more organs!",
      "color": "#6092C0",
      "categories": []
    }
  ]
}
```
</p>
</details> 

<details><summary>Local Storage Dump After Upgrade</summary>
<p>

```json 
{
  "elasticAssistantDefault.knowledgeBase": {
    "isEnabledRAGAlerts": false,
    "isEnabledKnowledgeBase": false,
    "latestAlerts": 20
  },
  "elasticAssistantDefault.systemPrompts": [
    {
      "id": "default-system-prompt",
      "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:",
      "name": "Default system prompt",
      "promptType": "system",
      "isDefault": true,
      "isNewConversationDefault": true
    },
    {
      "id": "CB9FA555-B59F-4F71-AFF9-8A891AC5BC28",
      "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nProvide the most detailed and relevant answer possible, as if you were relaying this information back to a cyber security expert.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:",
      "name": "Enhanced system prompt",
      "promptType": "system",
      "isDefault": true
    }
  ],
  "elasticAssistantDefault.quickPrompts": [
    {
      "title": "Alert summarization",
      "prompt": "As an expert in security operations and incident response, provide a breakdown of the attached alert and summarize what it might mean for my organization.",
      "color": "#F68FBE",
      "categories": [
        "alert"
      ],
      "isDefault": true
    },
    {
      "title": "ES|QL Query Generation",
      "prompt": "As an expert user of Elastic Security, please generate an accurate and valid ESQL query to detect the use case below. Your response should be formatted to be able to use immediately in an Elastic Security timeline or detection rule. Take your time with the answer, check your knowledge really well on all the functions I am asking for. For ES|QL answers specifically, you should only ever answer with what's available in your private knowledge. I cannot afford for queries to be inaccurate. Assume I am using the Elastic Common Schema and Elastic Agent.\\n\\nEnsure the answers are formatted in a way which is easily copyable as a separate code block in markdown.",
      "color": "#9170B8",
      "categories": [
        "knowledge-base"
      ],
      "isDefault": true
    },
    {
      "title": "Query generation",
      "prompt": "As an expert user of Elastic Security, please generate an accurate and valid EQL query to detect the use case below. Your response should be formatted to be able to use immediately in an Elastic Security timeline or detection rule. If Elastic Security already has a prebuilt rule for the use case, or a similar one, please provide a link to it and describe it.",
      "categories": [
        "detection-rules"
      ],
      "color": "#7DDED8",
      "isDefault": true
    },
    {
      "title": "Workflow suggestions",
      "prompt": "As an expert user of Elastic Security, please suggest a workflow, with step by step instructions on how to:",
      "color": "#36A2EF",
      "isDefault": true
    },
    {
      "title": "Custom data ingestion helper",
      "prompt": "As an expert user of Elastic Security, Elastic Agent, and Ingest pipelines, please list accurate and formatted, step by step instructions on how to ingest the following data using Elastic Agent and Fleet in Kibana and convert it to the Elastic Common Schema:",
      "categories": [
        "event"
      ],
      "color": "#F3D371",
      "isDefault": true
    },
    {
      "title": "Query conversion",
      "prompt": "I have the following query from a previous SIEM platform. As an expert user of Elastic Security, please suggest an Elastic EQL equivalent. I should be able to copy it immediately into an Elastic security timeline.",
      "color": "#BADA55",
      "isDefault": true
    },
    {
      "title": "Agent integration advice",
      "prompt": "Which Fleet enabled Elastic Agent integration should I use to collect logs and events from:",
      "color": "#FFA500",
      "isDefault": true
    }
  ],
  "securitySolution.securityAssistant.systemPrompts": [
    {
      "id": "default-system-prompt",
      "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:",
      "name": "Default system prompt",
      "promptType": "system",
      "isDefault": true,
      "isNewConversationDefault": false
    },
    {
      "id": "CB9FA555-B59F-4F71-AFF9-8A891AC5BC28",
      "content": "You are a helpful, expert assistant who answers questions about Elastic Security. Do not answer questions unrelated to Elastic Security.\\nProvide the most detailed and relevant answer possible, as if you were relaying this information back to a cyber security expert.\\nIf you answer a question related to KQL, EQL, or ES|QL, it should be immediately usable within an Elastic Security timeline; please always format the output correctly with back ticks. Any answer provided for Query DSL should also be usable in a security timeline. This means you should only ever include the 'filter' portion of the query.\\nUse the following context to answer questions:",
      "name": "Enhanced system prompt",
      "promptType": "system",
      "isDefault": true,
      "isNewConversationDefault": false
    },
    {
      "id": "New Pre-Persistence System Prompt",
      "content": "Body!",
      "name": "New Pre-Persistence System Prompt",
      "promptType": "system",
      "isNewConversationDefault": true
    }
  ],
  "securitySolution.securityAssistant.lastConversationId": "Alert summary",
  "securitySolution.securityAssistant.defaultAllow": [
    "_id",
    "@timestamp",
    "cloud.availability_zone",
    "cloud.provider",
    "cloud.region",
    "destination.ip",
    "dns.question.name",
    "dns.question.type",
    "event.action",
    "event.category",
    "event.dataset",
    "event.module",
    "event.outcome",
    "event.type",
    "file.Ext.original.path",
    "file.hash.sha256",
    "file.name",
    "file.path",
    "host.name",
    "host.risk.calculated_level",
    "host.risk.calculated_score_norm",
    "kibana.alert.last_detected",
    "kibana.alert.risk_score",
    "kibana.alert.rule.description",
    "kibana.alert.rule.name",
    "kibana.alert.rule.references",
    "kibana.alert.rule.threat.framework",
    "kibana.alert.rule.threat.tactic.id",
    "kibana.alert.rule.threat.tactic.name",
    "kibana.alert.rule.threat.tactic.reference",
    "kibana.alert.rule.threat.technique.id",
    "kibana.alert.rule.threat.technique.name",
    "kibana.alert.rule.threat.technique.reference",
    "kibana.alert.rule.threat.technique.subtechnique.id",
    "kibana.alert.rule.threat.technique.subtechnique.name",
    "kibana.alert.rule.threat.technique.subtechnique.reference",
    "kibana.alert.severity",
    "kibana.alert.workflow_status",
    "process.args",
    "process.command_line",
    "process.executable",
    "process.Ext.token.integrity_level_name",
    "process.entity_id",
    "process.exit_code",
    "process.hash.md5",
    "process.hash.sha1",
    "process.name",
    "process.hash.sha256",
    "process.parent.args",
    "process.parent.args_count",
    "process.parent.code_signature.exists",
    "process.parent.code_signature.status",
    "process.parent.code_signature.subject_name",
    "process.parent.code_signature.trusted",
    "process.parent.command_line",
    "process.parent.entity_id",
    "process.parent.executable",
    "process.pid",
    "process.working_directory",
    "network.protocol",
    "source.ip",
    "user.domain",
    "user.name",
    "user.risk.calculated_level",
    "user.risk.calculated_score_norm"
  ],
  "securitySolution.securityAssistant.defaultAllowReplacement": [
    "_id",
    "cloud.availability_zone",
    "cloud.provider",
    "cloud.region",
    "destination.ip",
    "file.Ext.original.path",
    "file.name",
    "file.path",
    "host.ip",
    "host.name",
    "source.ip",
    "user.domain",
    "user.name"
  ],
  "securitySolution.securityAssistant.knowledgeBase": {
    "isEnabledRAGAlerts": false,
    "isEnabledKnowledgeBase": false,
    "latestAlerts": 20
  },
  "securitySolution.securityAssistant.quickPrompts": [
    {
      "title": "Alert summarization",
      "prompt": "As an expert in security operations and incident response, provide a breakdown of the attached alert and summarize what it might mean for my organization.",
      "color": "#F68FBE",
      "categories": [
        "alert"
      ],
      "isDefault": true
    },
    {
      "title": "ES|QL Query Generation",
      "prompt": "As an expert user of Elastic Security, please generate an accurate and valid ESQL query to detect the use case below. Your response should be formatted to be able to use immediately in an Elastic Security timeline or detection rule. Take your time with the answer, check your knowledge really well on all the functions I am asking for. For ES|QL answers specifically, you should only ever answer with what's available in your private knowledge. I cannot afford for queries to be inaccurate. Assume I am using the Elastic Common Schema and Elastic Agent.\\n\\nEnsure the answers are formatted in a way which is easily copyable as a separate code block in markdown.",
      "color": "#9170B8",
      "categories": [
        "knowledge-base"
      ],
      "isDefault": true
    },
    {
      "title": "Query generation",
      "prompt": "As an expert user of Elastic Security, please generate an accurate and valid EQL query to detect the use case below. Your response should be formatted to be able to use immediately in an Elastic Security timeline or detection rule. If Elastic Security already has a prebuilt rule for the use case, or a similar one, please provide a link to it and describe it.",
      "categories": [
        "detection-rules"
      ],
      "color": "#7DDED8",
      "isDefault": true
    },
    {
      "title": "Workflow suggestions",
      "prompt": "As an expert user of Elastic Security, please suggest a workflow, with step by step instructions on how to:",
      "color": "#36A2EF",
      "isDefault": true
    },
    {
      "title": "Custom data ingestion helper",
      "prompt": "As an expert user of Elastic Security, Elastic Agent, and Ingest pipelines, please list accurate and formatted, step by step instructions on how to ingest the following data using Elastic Agent and Fleet in Kibana and convert it to the Elastic Common Schema:",
      "categories": [
        "event"
      ],
      "color": "#F3D371",
      "isDefault": true
    },
    {
      "title": "Query conversion",
      "prompt": "I have the following query from a previous SIEM platform. As an expert user of Elastic Security, please suggest an Elastic EQL equivalent. I should be able to copy it immediately into an Elastic security timeline.",
      "color": "#BADA55",
      "isDefault": true
    },
    {
      "title": "Agent integration advice",
      "prompt": "Which Fleet enabled Elastic Agent integration should I use to collect logs and events from:",
      "color": "#FFA500",
      "isDefault": true
    },
    {
      "title": "Custom Pre-Persistence QP",
      "prompt": "No wonder there's so much body, you've got an organ going!",
      "color": "#D36086",
      "categories": [
        "detection-rules"
      ]
    },
    {
      "title": "Custom Pre-Persistence ",
      "prompt": "More body, more organs!",
      "color": "#6092C0",
      "categories": []
    }
  ]
}
```
</p>
</details> 

---

I think ideally it'd be nice if we could remove all keys that are no longer needed, and so just end up with local storage state for `elasticAssistantDefault.lastConversationId` and `elasticAssistantDefault.knowledgeBase`. We could keep them around for a little while as backups, but maybe that's not necessary...?

Reply:
> That said, looks like replacements are being created and attached as expected in the conversations datastream.

Actually, there might be a slight bug here. I kept re-using the same conversation for testing, and looks like `replacements` are not cleared out when the conversation is cleared, and subsequent messages will just append to `replacements`. Everything works as expected though, but I think we'll want to clear out the replacements when the conversation is cleared (as the underlying messages the replacements reference no longer exist).

Reply:
Spoke w/ @YulNaumenko -- prompt/anonymization API's (and any migration) will be wired up in a follow-up PR.